AudioRecorder.py

```diff
@@ -37,23 +45,44 @@ def __init__(self):
         self.adjust_for_noise("Default Mic", "Please make some noise from the Default Mic...")
 
 class DefaultSpeakerRecorder(BaseRecorder):
-    def __init__(self):
-        with pyaudio.PyAudio() as p:
-            wasapi_info = p.get_host_api_info_by_type(pyaudio.paWASAPI)
-            default_speakers = p.get_device_info_by_index(wasapi_info["defaultOutputDevice"])
-            
-            if not default_speakers["isLoopbackDevice"]:
-                for loopback in p.get_loopback_device_info_generator():
-                    if default_speakers["name"] in loopback["name"]:
-                        default_speakers = loopback
-                        break
-                else:
+    
+    # Different implementations of obtaining the info dict of a default speaker, for different platforms
+    if os.name == "nt":
+        def _get_default_speaker(self):
+            # Requires PyAudioWPatch >= 0.2.12.6
+            with pyaudio.PyAudio() as p:
+                try:
+                    # Get loopback of default WASAPI speaker
+                    return p.get_default_wasapi_loopback()
+        
+                except OSError:                
+                    print("[ERROR] Looks like WASAPI is not available on the system.")
+        
+                except LookupError:
                     print("[ERROR] No loopback device found.")
+    
+    else:
+        def _get_default_speaker(self):
+            # At the moment, recording from speakers is only available under Windows
+            # raise NotImplementedError("Recording from speakers is only available under Windows")
+            
+            # As far as I understand, now the code style does not provide
+            # for error handling - only print them.
+            print("[ERROR] Recording from speakers is only available under Windows.")
+
+            
+    def __init__(self):
+        default_speaker = self._get_default_speaker()
+        
+        if not default_speaker:
+            print("[ERROR] Something went wrong while trying to get the default speakers.")
+            super().__init__(source=sr.Microphone(sample_rate=16000), source_name="Speaker")
```

Comment:
Don't think its appropriate to duplicate default mic here. Might be better to just have source as None and extend code to allow none sources where we do nothing with that recorder...not sure

Reply:
This is what I originally wanted to do, BUT for this it will be necessary to make a number of changes to the `main.py`, `AudioTranscriber.py` and possibly in other places, since the current logic does not provide for such interventions.

Again, if you think this is a necessary change, we can discuss it. I thought that rewriting a significant part of the code to create a temporary (until the issue with the speakers is resolved) patch is unnecessary.

Reply:
Don't think a rewrite is necessary, more of a extension of the code where we also allow no op recorder would be fine right? Only thing is AudioTranscriber init should be able to handle none source. Hopefully code is written where supporting MacOS with no op recording can be extended easily again to supporting MacOS with specific mac recording.

Reply:
@SevaSk  Any update here?